src/prefect/engine/cloud/flow_runner.py

```diff
@@ -393,15 +393,28 @@ def initialize_run(  # type: ignore
         tasks = {slug: t for t, slug in self.flow.slugs.items()}
         # update task states and contexts
         for task_run in flow_run_info.task_runs:
+
             try:
                 task = tasks[task_run.task_slug]
             except KeyError as exc:
                 msg = (
                     f"Task slug {task_run.task_slug} not found in the current Flow; "
-                    f"this is usually caused by changing the Flow without reregistering "
-                    f"it with the Prefect API."
+                    "this is usually caused by a mismatch between the serialized flow "
+                    "metadata stored by the Prefect API and the flow loaded from "
+                    f"storage. Did you change the flow without re-registering it?"
```

Comment:
👏

Reply:
Also note this doesn't need the `f` prefix (surprised flake8 didn't catch that).